spec/requests/grouped_content_and_links_spec.rb

```diff
@@ -0,0 +1,153 @@
+
+      expect(response.status).to eq(422)
+    end
+
+    it "422s when page_size is too large" do
+      get "/v2/grouped_content_and_links", page_size: 1001
+
+      expect(response.status).to eq(422)
+    end
+
+    it "422s when last_seen_content_id is not parsed as a uuid" do
+      get "/v2/grouped_content_and_links", last_seen_content_id: 'not a valid uuid', page_size: 1
+
+      expect(response.status).to eq(422)
+    end
+  end
+
+  context "batching by content_id" do
+    it "delivers all content items matching the content_ids contained in the batch" do
+      create_content_items('0001', '0002', '0003', '0005', '0007', '0008')
+
+      expected_result = [
+        hash_including(content_id: make_test_id('0005')),
+        hash_including(content_id: make_test_id('0007')),
+      ]
+
+      get "/v2/grouped_content_and_links", last_seen_content_id: make_test_id('0003'), page_size: 2
+
+      expect(response.status).to eq(200)
+      expect(results(response)).to match_array(expected_result)
+    end
+
+    it "delivers a partial batch when reaching the final content_id" do
+      create_content_items('0001', '0002', '0003', '0005', '0007', '0008')
+
+      expected_result = [
+        hash_including(content_id: make_test_id('0008')),
+      ]
+
+      get "/v2/grouped_content_and_links", last_seen_content_id: make_test_id('0007'), page_size: 2
+
+      expect(response.status).to eq(200)
+      expect(results(response)).to match_array(expected_result)
+    end
+
+    it "delivers an empty batch when reading beyond the final content_id" do
+      create_content_items('0001', '0002', '0003', '0005', '0007', '0008')
+
+      expected_result = []
+
+      get "/v2/grouped_content_and_links", last_seen_content_id: make_test_id('0008'), page_size: 2
+
+      expect(response.status).to eq(200)
+      expect(results(response)).to match_array(expected_result)
+    end
+  end
+
+  context "response format" do
+    it "delivers expected content fields" do
+      create_content_items('0001')
+      test_content_id = make_test_id('0001')
+      test_base_path = make_base_path(test_content_id)
+      links_hash = {
+        "topics" => [make_test_id('0002'), make_test_id('0003')],
+        "organisations" => [make_test_id('0004'), make_test_id('0005')],
+      }
+
+      FactoryGirl.create(
+        :link_set,
+        content_id: test_content_id,
+        links_hash: links_hash,
+      )
+
+      expected_result = [
+        {
+          content_id: test_content_id,
+          content_items: [
+            hash_including(
+              base_path: test_base_path,
+            )
+          ],
+          links: links_hash.symbolize_keys
+        }
+      ]
+
+
+      get "/v2/grouped_content_and_links", page_size: 1
+
+      expect(response.status).to eq(200)
+      expect(results(response)).to match_array(expected_result)
+    end
+  end
+
+private
+
+  def results(response)
+    JSON.parse(response.body, symbolize_names: true)[:results]
```

Comment:
You can replace this with `parsed_respons["results"]` (https://github.com/alphagov/publishing-api/pull/299)